core/src/scene/scene.h

```diff
@@ -28,12 +28,21 @@ struct Stops;
  * Scene is a singleton containing the styles, lighting, and interactions defining a map scene
  */
 
+#define COMPONENT_PATH_DELIMITER '.'
+
+struct UserDefinedSceneValue {
+    std::vector<std::string> splitPath;
+    std::string value;
+};
+
+using UserDefinedSceneValuePair = std::pair<std::string, UserDefinedSceneValue>;
```

Comment:
No actually this can go now, it was only used for user error messages but we can reconstruct it from the split path.